Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.cpp

```diff
@@ -551,6 +551,64 @@ void Wiimote::Update(const WiimoteEmu::DesiredWiimoteState& target_state)
   SendDataReport(target_state);
 }
 
+bool Wiimote::IsProcessingEncryptedExtension(
+    const WiimoteCommon::DataReportBuilder& rpt_builder) const
+{
+  return rpt_builder.HasExt() && m_active_extension > ExtensionNumber::NONE &&
+         m_active_extension < ExtensionNumber::MAX;
+}
+
+void Wiimote::EncryptExtension(WiimoteCommon::DataReportBuilder& rpt_builder)
+{
+  u8* ext_data = rpt_builder.GetExtDataPtr();
+  WiimoteEmu::EncryptionKey encryption_key = GetExtensionEncryptionKey();
+
+  switch (m_active_extension)
```

Comment:
I updated my code to remove these switch statements, since they were unnecessary.

Although, I was wondering - is it ever possible for an extension to be plugged in which isn't encrypted? Is MotionPlus the only case where that happens? And if you call Encrypt/Decrypt with an encryption key of all 0s, does it change the input?

Reply:
It is possible for an extension to be used without encryption, but games do not do this in practice.

The Encrypt/Decrypt functions are just addition/subtraction/xor, which all have no effect with 0s, so the data will not change in that case.